cmd/minikube/cmd/delete.go

```diff
@@ -21,18 +21,27 @@ import (
 	"os"
 
 	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
 	cmdUtil "k8s.io/minikube/cmd/util"
 	"k8s.io/minikube/pkg/minikube/cluster"
+	"k8s.io/minikube/pkg/minikube/config"
 	"k8s.io/minikube/pkg/minikube/machine"
 )
 
 // deleteCmd represents the delete command
 var deleteCmd = &cobra.Command{
-	Use:   "delete",
-	Short: "Deletes a local kubernetes cluster",
-	Long: `Deletes a local kubernetes cluster. This command deletes the VM, and removes all
-associated files.`,
+	Use:   "delete MINIKUBE_PROFILE_NAME",
```

Comment:
I don't think we want to break the default `minikube delete` case.

As a fix for #1683, I think we should just make sure to error on unknown arguments after command, rather than silently ignore them. i.e., ` $ minikube delete arg1 arg2` should fail, since we shouldn't ignore unknown arguments

As far as profiles go, each command accepts the `--profile` flag. 

` $ minikube delete`, should work and delete the default machine
` $ minikube delete --profile vm2` should delete the minikube profile vm2

I'm not a huge fan of moving the profile flag behavior to an optional noun after `minikube <start, stop, delete>` since we'd probably have to have every command behave that way.  Sorry if my original comment on the issue was unclear,
> minikube delete either takes no args or just the machine name

I think it really can't take the machine name without a pretty big change to the UX (since all commands should behave similarly.)  But the enforcing the former (no unknown args) is definitely something we should do. 

Let me know if that makes sense